pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/task"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/failureinjection/failures"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/errors"
+)
+
+type failureSmokeTest struct {
+	testName    string
+	failureName string
+	args        failures.FailureArgs
+
+	// Time to wait for the failure to propagate after inject/restore.
+	waitForFailureToPropagate time.Duration
+	// Validate that the failure was injected correctly, called after Setup() and Inject().
+	validateFailure func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error
+	// Validate that the failure was restored correctly, called after Restore().
+	validateRestore func(ctx context.Context, l *logger.Logger, c cluster.Cluster) error
+}
+
+func (t *failureSmokeTest) run(
+	ctx context.Context, l *logger.Logger, c cluster.Cluster, fr *failures.FailureRegistry,
+) error {
+	// TODO(darryl): In the future, roachtests should interact with the failure injection library
+	// through helper functions in roachtestutil so they don't have to interface with roachprod
+	// directly.
+	failure, err := fr.GetFailure(c.MakeNodes(), t.failureName, l, c.IsSecure())
```

Comment:
Nit: `GetFailureMode` and `failureMode` to disambiguate? Seeing both `failure` and `err` in the same scope can be confusing.